### PR TITLE
fix(1319): add build cluster model

### PR DESCRIPTION
**Pull request by minz1027**
- Add build cluster model
- Add buildClusterName to build

Related to https://github.com/screwdriver-cd/screwdriver/issues/1319

### CODE REVIEW

File: models/buildCluster.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+const Joi = require('joi');
+const mutate = require('../lib/mutate');
+const Command = require('../config/command');
+const scmContext = Joi.reach(require('./pipeline').base, 'scmContext');
+const scmOrgization = Joi
```
Comment (tkyi) [inline models/buildCluster.js]: scmOrg`an`ization*

File: models/buildCluster.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        .example('iOS'),
+
+    description: Joi
+        .string().max(100)
+        .description('Description of the build cluster')
+        .example('Build cluster for iOS team'),
+
+    scmContext,
+
+    scmOrganizations: Joi.array().items(scmOrgization),
```
Comment (tkyi) [inline models/buildCluster.js]: scmOrg`an`ization*

File: models/buildCluster.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    isActive: Joi
+        .boolean()
+        .description('Flag if the the build cluster is active')
+        .example(true)
+        .default(true),
+
+    managedBy: Joi
```
Comment (tkyi) [inline models/buildCluster.js]: `maintainer` for consistency? not sure if that's too vague or not

File: models/buildCluster.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    description: Joi
+        .string().max(100)
+        .description('Description of the build cluster')
+        .example('Build cluster for iOS team'),
+
+    scmContext,
+
+    scmOrganizations: Joi.array().items(scmOrgization),
+
+    isActive: Joi
```
Comment (tkyi) [inline models/buildCluster.js]: is this necessary?

File: models/buildCluster.js (hunk 1 above)
Comment (minz1027) [reply models/buildCluster.js]: lolll

File: models/buildCluster.js (hunk 3 above)
Comment (minz1027) [reply models/buildCluster.js]: this is not the specific maintainer, it's more like the maintainer type? can either be screwdriver or external

File: models/buildCluster.js (hunk 4 above)
Comment (minz1027) [reply models/buildCluster.js]: yeah, e.g. when we have 2 clusters and want to temporarily turn off one for maintainence

File: models/buildCluster.js (hunk 3 above)
Comment (tkyi) [reply models/buildCluster.js]: ah I see
interesting

File: models/buildCluster.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+     * @type {Joi}
+     */
+    get: Joi.object(mutate(MODEL, [
+        'id', 'name', 'scmContext', 'scmOrganizations', 'isActive', 'managedBy', 'managedByEmail'
+    ], [
+        'description'
+    ])).label('Get BuildCluster'),
+
+    /**
+     * Properties for BuildCluster that will come back during a UPDATE request
```
Comment (tkyi) [inline models/buildCluster.js]: Properties for BuildCluster that will ~come back~`be passed` during a UPDATE request

File: models/buildCluster.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+     *
+     * @property update
+     * @type {Joi}
+     */
+    update: Joi.object(mutate(MODEL, [], [
+        'description', 'isActive', 'scmOrganizations', 'managedByEmail'
```
Comment (tkyi) [inline models/buildCluster.js]: should we be able to update the `name` and `managedBy` as well?

Comment (minz1027) [reply models/buildCluster.js]: fixed!

File: models/buildCluster.js (hunk 3 above)
Comment (minz1027) [reply models/buildCluster.js]: I agree it's better if it's consistent. @parthasl is working on the naming :D lol

Comment (tkyi) [reply models/buildCluster.js]: if it's only two possibilities, a boolean like `default` might make more sense for this

File: models/buildCluster.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+    isActive: Joi
+        .boolean()
+        .description('Flag if the the build cluster is active')
+        .example(true)
+        .default(true),
+
+    defaultCluster: Joi
+        .boolean()
+        .description('Flag the defulat cluster when user does not specify a build cluster')
```
Comment (tkyi) [inline models/buildCluster.js]: default*

**Comment by parthasl** [comment]
👍

**Pull request merged**